src/phonebook/Main.java

```diff
@@ -1,7 +1,72 @@
 package phonebook;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
 public class Main {
-    public static void main(String[] args) {
-        System.out.print("Hello world!");
+
+  static final String CONTACT_DATA_DIRECTORY = "data/directory.txt";
+  static final String FIND_REQUESTS_DIRECTORY = "data/find.txt";
+
+  public static void main(String[] args) throws IOException {
+    List<Contact> contacts = parseContacts();
+    bubbleSort(contacts);
+
+    List<String> requests = Files.readAllLines(Paths.get(FIND_REQUESTS_DIRECTORY));
+
+    long start = System.currentTimeMillis();
+    int cnt = 1;
+    for (String request : requests) {
+      Contact contact;
+      if ((contact = search(contacts, request)) != null) {
+        System.out.printf("%d. %s has number %s\n", cnt++, request, contact.getNumber());
+      }
     }
+    System.out.printf("To find all entries it taken %d ms\n", System.currentTimeMillis() - start);
+  }
+
+  private static Contact search(List<Contact> contacts, String name) {
+    int sqr = (int) Math.sqrt(contacts.size());
+    int lo = 0, hi = Math.min(sqr, contacts.size() - 1);
+    while (lo < hi) {
+      Contact contact = contacts.get(hi);
+      if (contact.getName().compareTo(name) == 0) {
```

Comment:
The utility class Objects has a set of convenient methods which provides null-safety instead of object.method1().method2() and so on. And they are more readable as well.

Reply:
Thanks I think it is reasonable to use Objects here. And think if java.util.Optional; class is appropriate to return from search method?